src/content/app/genome-browser/state/browser-entity/browserEntitySlice.ts

```diff
@@ -0,0 +1,219 @@
+
+import browserStorageService from 'src/content/app/genome-browser/services/browser-storage-service';
+import trackPanelStorageService from 'src/content/app/genome-browser/components/track-panel/services/track-panel-storage-service';
+
+import { fetchEnsObject } from 'src/shared/state/ens-object/ensObjectActions';
+import {
+  ChrLocation,
+  deleteGenomeLocationData,
+  setLocationDataFromUrl
+} from 'src/content/app/genome-browser/state/browser-location/browserLocationSlice';
+import {
+  deleteGenomeTrackPanelData,
+  setTrackPanelDataFromUrl
+} from 'src/content/app/genome-browser/state/track-panel/trackPanelSlice';
+import {
+  getBrowserActiveEnsObjectIds,
+  getBrowserTrackStates,
+  getBrowserActiveGenomeId
+} from 'src/content/app/genome-browser/state/browser-entity/browserEntitySelectors';
+import { ensureSpeciesIsEnabled } from 'src/content/app/species-selector/state/speciesSelectorSlice';
+
+import {
+  BrowserTrackStates,
+  TrackActivityStatus
+} from 'src/content/app/genome-browser/components/track-panel/trackPanelConfig';
+import { RootState } from 'src/store';
+
+export type BrowserEntityState = Readonly<{
+  activeGenomeId: string | null;
+  activeEnsObjectIds: { [genomeId: string]: string };
+  trackStates: BrowserTrackStates;
+}>;
+
+const isServer = typeof window === 'undefined';
+
+const activeGenomeId = !isServer
+  ? browserStorageService.getActiveGenomeId()
+  : null;
+const activeEnsObjectIds = !isServer
+  ? browserStorageService.getActiveEnsObjectIds()
+  : [];
+const trackStates = !isServer ? browserStorageService.getTrackStates() : {};
+
+export const defaultBrowserEntityState: BrowserEntityState = {
+  activeGenomeId,
+  activeEnsObjectIds,
+  trackStates
+};
+
+export type ParsedUrlPayload = {
+  activeGenomeId: string;
+  activeEnsObjectId: string | null;
+  chrLocation: ChrLocation | null;
+};
+
+export type UpdateTrackStatesPayload = {
+  genomeId: string;
+  categoryName: string;
+  trackId: string;
+  status: TrackActivityStatus;
+};
+
+export const fetchDataForLastVisitedObjects: ActionCreator<
+  ThunkAction<void, any, null, Action<string>>
+> = () => async (dispatch, getState: () => RootState) => {
+  const state = getState();
+  const activeEnsObjectIdsMap = getBrowserActiveEnsObjectIds(state);
+  Object.values(activeEnsObjectIdsMap).forEach((objectId) =>
+    dispatch(fetchEnsObject(objectId))
+  );
+};
+
+export const updateTrackStatesAndSave: ActionCreator<
+  ThunkAction<void, any, null, Action<string>>
+> = (payload: BrowserTrackStates) => (dispatch, getState: () => RootState) => {
+  dispatch(updateTrackStates(payload));
+  const trackStates = getBrowserTrackStates(getState());
+  browserStorageService.saveTrackStates(trackStates);
+};
+
+export const setDataFromUrlAndSave: ActionCreator<
+  ThunkAction<void, any, null, Action<string>>
+> = (payload: ParsedUrlPayload) => (dispatch) => {
+  dispatch(setEntityDataFromUrl(payload));
+  dispatch(setLocationDataFromUrl(payload));
```

Comment:
Is there any reason to separate `browserEntitySlice` from `browserLocation` slice? They seem to have very closely related responsibilities. There doesn't seem to be any rationale in having `activeEnsObjectIds` and `trackStates` in this slice, yet having locations in a different slice.

Reply:
That's all done